packages/cli/src/evaluation.ee/test-runner/test-runner.service.ee.ts

```diff
@@ -135,11 +222,45 @@ export class TestRunnerService {
 	 * Get the evaluation metrics nodes from a workflow.
 	 */
 	static getEvaluationMetricsNodes(workflow: IWorkflowBase) {
-		return workflow.nodes.filter((node) => node.type === EVALUATION_METRICS_NODE);
+		return workflow.nodes.filter(
+			(node) => node.type === EVALUATION_NODE && node.parameters.operation === 'setMetrics',
+		);
+	}
+
+	/**
+	 * Extract the dataset trigger output
+	 */
+	private extractDatasetTriggerOutput(execution: IRun, workflow: IWorkflowBase) {
+		const triggerNode = this.findTriggerNode(workflow);
+		assert(triggerNode);
+
+		const triggerOutputData = execution.data.resultData.runData[triggerNode.name][0];
+		const triggerOutput = triggerOutputData?.data?.main?.[0];
+
+		if (!triggerOutput || triggerOutput.length === 0) {
+			throw new TestRunError('TEST_CASES_NOT_FOUND');
+		}
+
+		return triggerOutput;
+	}
+
+	/**
+	 * Evaluation result is collected from all Evaluation Metrics nodes
+	 */
+	private extractEvaluationResult(execution: IRun, workflow: IWorkflowBase): IDataObject {
+		// TODO: Do not fail if not all metric nodes were executed
+		const metricsNodes = TestRunnerService.getEvaluationMetricsNodes(workflow);
+		const metricsRunData = metricsNodes.flatMap(
+			(node) => execution.data.resultData.runData[node.name],
+		);
+		const metricsData = metricsRunData.reverse().map((data) => data.data?.main?.[0]?.[0]?.json);
+		const metricsResult = metricsData.reduce((acc, curr) => ({ ...acc, ...curr }), {}) ?? {};
```

Comment:
niptick - does not seem needed
```suggestion
		const metricsResult = metricsData.reduce((acc, curr) => ({ ...acc, ...curr }), {});
```